saleor/payment/gateways/adyen/tests/webhooks/test_handle_notifications.py

```diff
@@ -1010,3 +1011,37 @@ def test_handle_cancel_or_refund_action_cancel_invalid_payment_id(
     handle_cancel_or_refund(notification, config)
 
     assert f"Unable to decode the payment ID {invalid_reference}." in caplog.text
+
+
+@pytest.mark.parametrize("return_value", (True, False))
+@mock.patch("saleor.payment.models.Payment.can_create_order")
+@mock.patch("saleor.payment.gateways.adyen.webhooks.create_order")
+def test_handle_not_created_order(
+    mock_create_order,
+    mock_can_create_order,
+    return_value,
+    notification,
+    payment_adyen_for_checkout,
+):
+    mock_can_create_order.return_value = return_value
+    payment = payment_adyen_for_checkout
+    payment_id = graphene.Node.to_global_id("Payment", payment.pk)
+
+    notification = notification(
+        merchant_reference=payment_id,
+        value=price_to_minor_unit(payment.total, payment.currency),
+    )
+    manager = get_plugins_manager()
+
+    handle_not_created_order(
+        notification,
+        payment,
+        payment.checkout,
+        TransactionKind.ACTION_TO_CONFIRM,
+        manager,
+    )
+
+    if return_value is True:
```

Comment:
I think it's not a good idea to create any logic in the tests. It's hard to mantain and understand. How about to create to separate tests one for ` mock_create_order.assert_called_once_with(payment, payment.checkout, manager)` and
second one for `mock_create_order.assert_not_called()`?

Reply:
I decided to go that way as the difference in the tests is tiny and the logic isn't complicated. But no problem with splitting it into multiple tests. Done.